### PR TITLE
correct NU3024 message for validation scenario

### PR DESCRIPTION
**Pull request by heng-liu**
Bug
Fixes: NuGet/Home/[#7933](https://github.com/NuGet/Home/issues/7933)
Regression: No
Fix
Correct NU3024 message for two scenarios.
In signing scenario, change the message string to make it more precise.
In validation scenario, add a new error message string, as it should not be the same with NU3022.
Testing/Validation
Tests Added: No

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/Signing/Utility/VerificationUtility.cs [hunk 1]
```diff
@@ -144,7 +144,7 @@ internal static SignatureVerificationStatusFlags ValidateTimestamp(Timestamp tim
 
                 if (!spec.AllowedHashAlgorithmOids.Contains(signerInfo.DigestAlgorithm.Value))
                 {
-                    issues.Add(SignatureLog.Issue(treatIssuesAsErrors, NuGetLogCode.NU3024, string.Format(CultureInfo.CurrentCulture, Strings.VerifyError_TimestampUnsupportedSignatureAlgorithm, signature.FriendlyName)));
+                    issues.Add(SignatureLog.Issue(treatIssuesAsErrors, NuGetLogCode.NU3024, string.Format(CultureInfo.CurrentCulture, Strings.VerifyError_TimestampUnsupportedDigestAlgorithm, signature.FriendlyName)));
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging/Signing/Utility/VerificationUtility.cs]: You are passing the signature.FriendlyName here but there's no placeholder for it in the message.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 2]
```diff
@@ -227,6 +227,9 @@
     <value>The {0}'s timestamp certificate has an unsupported signature algorithm.</value>
     <comment>0 - Signature friendly name</comment>
   </data>
+  <data name="VerifyError_TimestampUnsupportedDigestAlgorithm" xml:space="preserve">
+    <value>The timestamp signature has an unsupported digest algorithm.</value>
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx]: Follow the example of the above resource string (`VerifyError_TimestampUnsupportedSignatureAlgorithm`):

`The {0}'s timestamp signature has an unsupported digest algorithm.`

That way, the full error message will be one of the following:

* The author primary signature's timestamp signature has an unsupported digest algorithm.
* The repository primary signature's timestamp signature has an unsupported digest algorithm.
* The repository countersignature's timestamp signature has an unsupported digest algorithm.

File: src/NuGet.Core/NuGet.Packaging/Signing/Utility/VerificationUtility.cs (hunk 1 above)
Comment (heng-liu) [reply src/NuGet.Core/NuGet.Packaging/Signing/Utility/VerificationUtility.cs]: Thanks! I'll fix it.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx (hunk 2 above)
Comment (heng-liu) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx]: Thanks for your explanation! I'll fix it.

**Review by dtivel — APPROVED** [review]

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 3]
```diff
@@ -227,6 +227,10 @@
     <value>The {0}'s timestamp certificate has an unsupported signature algorithm.</value>
     <comment>0 - Signature friendly name</comment>
   </data>
+  <data name="VerifyError_TimestampUnsupportedDigestAlgorithm" xml:space="preserve">
+    <value>The {0}'s timestamp signature has an unsupported digest algorithm.</value>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx:231]: Should this be enhanced by calling out the name of the digest algorithm? 

@dtivel

Comment (dtivel) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx:231]: I think the OID is sufficient.

Comment (dtivel) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx:231]: There are lots of places where we could do this, so I'm OK with making this change here and everywhere in a separate PR.  But open a separate issue for it.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx:231]: Created an issue for that https://github.com/NuGet/Home/issues/7926

**Review by nkolev92 — APPROVED** [review]
👏

File: src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs [hunk 4]
```diff
@@ -361,7 +361,7 @@ public enum NuGetLogCode
         NU3023 = 3023,
 
         /// <summary>
-        /// The timestamp signing certificate has an unsupported signature algorithm.
+        /// The timestamp signature has an unsupported signature algorithm.
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs]: The timestamp signature has an unsupported digest algorithm.

**Review by dtivel — APPROVED** [review]
I think your change will break [this test](https://github.com/NuGet/NuGet.Client/blob/8d6b7ff631d103b2a2005ccd8b98aca642ccff8e/test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/SignCommandTests.cs#L710).  Make sure your test results are clean before merging.

**Comment by heng-liu** [comment]
Created another PR instead: https://github.com/NuGet/NuGet.Client/pull/2786

**Pull request closed without merging**